src/cli/generate-idlparsed.js

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env node
+/**
+ * The parsed IDL generator takes a crawl report or a single spec as input, and
+ * generates (or re-generates if it already exists) a parsed IDL structure from
+ * the raw IDL that the spec defines. Result is dumped to the console or saved
+ * to the given folder.
+ * 
+ * The parsed IDL generator is used by the crawler to create and save the parsed
+ * IDL structures. It is also useful to re-generated the parsed IDL info when
+ * an IDL patch has been applied to the raw IDL.
+ *
+ * The parsed IDL generator can be called directly through:
+ *
+ * `node generate-idlparsed.js [crawl report] [save folder]`
+ *
+ * where `crawl report` is the path to the folder that contains the
+ * `index.json` file and all other crawl results produced by specs-crawler.js,
+ * and `save folder` is an optional folder (which must exist) where IDL
+ * name extracts are to be saved. In the absence of this parameter, the report
+ * is written to the console.
+ *
+ * When a folder is provided, the IDL name extracts are saved as a JSON
+ * structure in an `idlparsed` subfolder.
+ */
+
+const fs = require('fs');
+const path = require('path');
+const webidlParser = require('../cli/parse-webidl');
+const {
+  expandCrawlResult,
+  requireFromWorkingDirectory
+} = require('../lib/util');
+
+
+/**
+ * Update the spec object in place with parsed IDL information.
+ * 
+ * @function
+ * @public
+ * @param {Object} spec The spec object to update. The function looks for the
+ *   raw IDL in the `idl` property.
+ * @return {Object} The updated spec with an `idl` property that contains the
+ *   parsed version of the IDL, and the raw IDL moved under the `idl.idl`
+ *   sub-property. Note the spec object is updated in place.
+ */
+async function generateIdlParsed(spec) {
+  if (!spec?.idl) {
+    return spec;
+  }
+  const rawIdl = spec.idl.idl ?? spec.idl;
+  try {
+    const parsedIdl = await webidlParser.parse(rawIdl);
+    parsedIdl.hasObsoleteIdl = webidlParser.hasObsoleteIdl(rawIdl);
+    parsedIdl.idl = rawIdl;
+    spec.idl = parsedIdl;
+  }
+  catch (err) {
+    // IDL content is invalid and cannot be parsed.
+    // Let's return the error, along with the raw IDL
+    // content so that it may be saved to a file.
+    err.idl = rawIdl;
+    spec.idl = err;
+  }
+  return spec;
+}
+
+
+async function generateIdlParsedFromPath(crawlPath) {
+  const crawlIndex = requireFromWorkingDirectory(path.resolve(crawlPath, 'index.json'));
+  const crawlResults = await expandCrawlResult(crawlIndex, crawlPath, ['idl']);
+  await Promise.all(crawlResults.results.map(generateIdlParsed));
+  return crawlResults;
+}
+
+
+async function createFolderIfNeeded(name) {
```

Comment:
we have this [in generate-idlnames.js](https://github.com/w3c/reffy/blob/ef75e94483e8238b9c6879a5d48b9e60942c5dac/src/cli/generate-idlnames.js#L311) - it could be factored out in util.js?

Reply:
Done. Further factoring of the save functions could be envisioned but well, it ain't broken...